resources/uco_design_document.md

```diff
@@ -237,11 +297,21 @@ For instance, a digital photograph is represented as an ```observable:RasterPict
-
-If particular characterizing properties are directly relevant to the object across most use cases they are typically defined as properties directly on the class/object but where they may be characterizing a particular aspect of an object relevant to some but not all use cases they are typically defined as a facet that can be applied to the class/object when appropriate.
-
-For example, the properties ```core:source``` and ```core:target``` are always relevant to the ```core:Relationship``` class/object in all use cases and are therefore defined as direct properties of ```core:relationship```. On the other hand, differing use cases may characterize a ```location:Location``` as a simple address, or as a set of latitude-longitude coordinates, or as a set of GPS coordinates or any combination thereof. Because of this variation, properties characterizing a location in the form of an address are defined using the ```location:SimpleAddressFacet```, properties characterizing a location in the form of a set of latitude-longitude coordinates are defined using the ```location:LatLongCoordinatesFacet```, and a location in the form of a set of GPS coordinates are defined using the ```location:GPSCoordinatesFacet```.
+The generalized approach of facets on domain objects provides value by mitigating the above limitations/downsides as well as supporting many of the necessary use cases.
+
+* Simple support for use of particular non-domain concept classes across multiple domain concept classes
+* Simplified 3rd party extension of UCO classes/objects
+* Reduces verbosity and complexity for appropriate use cases
+* Significantly reduces shape churn for appropriate cases
+* Supports duck-typing use case (described in #5.1 below) very simply
+* Supports differential mechanism case (e.g., different mechanisms for expressing location info) simply without the complexity found in some ontologies taking a purely purpose-specific object property approach (e.g., GeoSPARQL)
+* Supports differential relevance case (e.g., different aspects of Identity such as birthinfo, credentials, citizenship info, physical characteristics, etc) very simply
+* Highly useful in adorning Relationships with various contextual information
+* Approach has been proven in CybOX, STIX and also recently incorporated into SPDX
+
+If particular characterizing properties are directly relevant to the object across most use cases they are typically defined as purpose-specific properties directly on the class/object but where they may be characterizing a particular aspect of an object relevant to some but not all use cases they are typically defined as a facet that can be applied to the class/object when appropriate.
```

Comment:
I request a more rigorous explanation for why a property would go on a `Facet` vs. on a `UcoObject`.  There is a lot of looseness in this description.

I can't use this to understand e.g. why `action:Action` and `tool:Tool` don't have corresponding `Facet`s.  The explanation for doing away with `ActionReferencesFacet` a while ago was that it was originally implemented for some use case that didn't pan out.  Nevermind whether a use case existed or not - why does `action:performer` get directly attached to an `action:Action`, whereas `observable:from` gets attached to an `observable:EmailMessageFacet`?